lib/SILGen/SILGenApply.cpp

```diff
@@ -5185,12 +5185,20 @@ ArgumentSource AccessorBaseArgPreparer::prepareAccessorAddressBaseArg() {
   if (shouldLoadBaseAddress()) {
     if (selfParam.isConsumed() ||
         base.getType().isAddressOnly(SGF.F)) {
-      // The load can only be a take if the base is a +1 rvalue.
-      auto shouldTake = IsTake_t(base.hasCleanup());
+      // Unlike arbitrary methods which can modify self via a capture,
+      // *implicit* accessors have no way to modify self, so self is immutable
+      // within the call. This means we don't have to emit a copy of the self
+      // parameter.
+      if (!selfParam.isIndirectInGuaranteed() ||
+          !cast<AccessorDecl>(accessor.getDecl())->isImplicit()) {
```

Comment:
I'm not sure SILGen should have any knowledge of which accessors are implicit and which ones are not. Some implicit accessors can have rather complicated bodies, for example lazy getters, or observer setters. In fact, the comment as written is wrong I believe; a didSet or willSet can capture `self`, even though the setter that calls it is implicit.

Can we instead check if the accessor's lowered capture list captures 'self'?

Reply:
That is a good point. What if we just checked if the accessor's self-access kind is non-mutating? Because both `didSet` and `setter` methods should have mutating self-access kinds, right? (Which, if I understand correctly, is basically the same logic you suggested.)

Reply:
Actually, I didn't really think this through. This will have the same issue as the original solution wrt exclusivity (or the lack thereof) if we remove the implicit check. 

And it sounds like it probably was still an issue because implicit accessors can still call non-implicit accessors which means "implicit" doesn't really give us any guarantees after all.